File: app/src/main/java/uk/co/victoriajanedavis/chatapp/data/mappers/TokenSpEntityMapper.java
package uk.co.victoriajanedavis.chatapp.data.mappers;

import uk.co.victoriajanedavis.chatapp.data.model.sharedpref.TokenSpModel;
import uk.co.victoriajanedavis.chatapp.domain.common.Mapper;
import uk.co.victoriajanedavis.chatapp.domain.entities.TokenEntity;

public class TokenSpEntityMapper extends Mapper<TokenSpModel, TokenEntity> {

    @Override
    public TokenEntity mapFrom(TokenSpModel from) {
        TokenEntity entity = new TokenEntity();

        entity.setToken(from.getToken());
        entity.setUserUuid(from.getUserUuid());
        entity.setUserUsername(from.getUserUsername());
        entity.setUserEmail(from.getUserEmail());

        return entity;
    }
}

File: app/src/main/java/uk/co/victoriajanedavis/chatapp/data/mappers/TokenSpEntityMapper.kt
package uk.co.victoriajanedavis.chatapp.data.mappers

import uk.co.victoriajanedavis.chatapp.data.model.sharedpref.TokenSpModel
import uk.co.victoriajanedavis.chatapp.domain.common.Mapper
import uk.co.victoriajanedavis.chatapp.domain.entities.TokenEntity

class TokenSpEntityMapper : Mapper<TokenSpModel, TokenEntity>() {

    override fun mapFrom(from: TokenSpModel): TokenEntity {
        return TokenEntity(
            token = from.token,
            userUuid = from.userUuid,
            userUsername = from.userUsername,
            userEmail = from.userEmail
        )
    }
}
